### PR TITLE
[QPG] Move naming qpg6100 to qpg

### LINKED ISSUE #6202: Rename qpg6100 in qpg for further extension
Rename QPG6100 naming throughout code tree to facilitate re-use of platform integration for further Qorvo chipsets

### PR DESCRIPTION
**Pull request by tima-q**
#### Problem
Qorvo Matter integration is initially done on the QPG6100 platform.
To prepare for wider integration across other chip-sets a renaming is to be done into the more generic 'qpg' prefix

#### Change overview
* Update all qpg6100 references into qpg.
* Introduce selection mechanism for other chip-sets
* Current example applications remain building for QPG6100.

#### Testing
* All QPG6100 based applications re-built manually and through GH CI
* GH CI ran to check possible impact on other build types

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

**Comment by woody-apple** [comment]
@saurabhst @msandstedt ?

**Review by saurabhst — APPROVED** [review]

File: third_party/qpg_sdk/qpg_sdk.gni [hunk 1]
```diff
@@ -16,31 +16,44 @@ import("//build_overrides/chip.gni")
...
+  qpg_sdk_root = qpg_sdk_root_env
 }
+assert(qpg_sdk_root != "", "qpg_sdk_root must be specified")
 
-assert(qpg6100_sdk_root != "", "qpg6100_sdk_root must be specified")
+qpg_target_ic_env = getenv("QPG_TARGET_IC")
+if (qpg_target_ic_env != "") {
+  qpg_target_ic = qpg_target_ic_env
+}
+assert(qpg_target_ic != "", "qpg_target_ic must be specified")
```
Comment (mspang) [inline third_party/qpg_sdk/qpg_sdk.gni:39]: Please make the explicit arguments override the environment, not the opposite. It's going to be terribly confusing if builddir/args.gn contains a value that is thrown away in favor of an environment variable.

I would actually recommend to remove the environment value handling entirely, and force configuration into the command line or build arguments, but I do understand the convenience it provides (the problems are two: (1) ninja won't automatically regenerate on environment changes so use of the environment can cause incorrect incremental build outputs and (2) environment variables, being cross-process global variables, have all the problems of global variables such as implicit coupling).

**Review by mspang — CHANGES_REQUESTED** [review]

File: third_party/qpg_sdk/qpg_sdk.gni (hunk 1 above)
Comment (tima-q) [reply third_party/qpg_sdk/qpg_sdk.gni:30]: @mspang Thanks for the feedback. The main intent here was to swap out the internal SDK easily during development. The arguments cover that purpose equally.
Removed the env overrides in favor of using the arguments.

**Review by jmartinez-silabs — APPROVED** [review]

**Review by mspang — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Comment by woody-apple** [comment]
@tima-q Can you resolve the conflicts here?

**Comment by tima-q** [comment]
> @tima-q Can you resolve the conflicts here?
Done ✅

**Comment by tima-q** [comment]
Fixes #6202

**Pull request merged**